llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1949,6 +1952,23 @@ bool SPIRVInstructionSelector::selectSign(Register ResVReg,
   return Result;
 }
 
+bool SPIRVInstructionSelector::selectWaveActiveAnyTrue(Register ResVReg,
+                                                       const SPIRVType *ResType,
+                                                       MachineInstr &I) const {
+  assert(I.getNumOperands() == 3);
+  assert(I.getOperand(2).isReg());
+
+  MachineBasicBlock &BB = *I.getParent();
+  SPIRVType *IntTy = GR.getOrCreateSPIRVIntegerType(32, I, TII);
+
+  return BuildMI(BB, I, I.getDebugLoc(), TII.get(SPIRV::OpGroupNonUniformAny))
```

Comment:
I'm looking at `selectWaveReadLaneAt` and they share a lot of code. Maybe we should look to do a more general selection emitter like for spirv opcodes the way that `selectExtInst` does it for opencl or opengl  extension opcodes?

Reply:
If it is easy to write a generic with a different number of operands then we could also look at the other wave intriniscs, eg `wave_is_first_lane` and see if they are also able to be included. They are all `SPIRV::Subgroup` ops.

Reply:
> If it is easy to write a generic with a different number of operands then we could also look at the other wave intriniscs

This should be farily reasonable to do. I'll take a look

Reply:
Okay, done! Please take a look!

Reply:
https://github.com/llvm/llvm-project/pull/115902/commits/8733a1b619dff0c8bcf86316c02252f2c4f5a6b6